source/develop/release-management/features/storage/incremental-backup.html.md

```diff
@@ -0,0 +1,181 @@
+---
+title: Incremental Backup
+category: feature
+authors: nsoffer, derez
+feature_name: Incremental Backup
+feature_modules: imageio,engine
+feature_status: Design
+---
+
+# Incremental Backup
+
+
+## Summary
+
+Incremental backup allows a user to backup only the changed blocks in
+all disks belonging to a VM snapshot. The typical use case is backing
+up the current changes in the active snapshot. Another use case may be
+backing up the previous snapshot if a user added a snapshot since the
+last backup.
+
+
+## Owner
+
+*   Nir Soffer (<nsoffer@redhat.com>)
+*   Daniel Erez (<derez@redhat.com>)
+
+
+## Incremental backup flow
+Backup flow is the same from the point of view of the user (backup
+application), but will be implemented differently under the hood.
+
+1. Backup application starts backup using oVirt API, specifying the last
+backup. The backup will include blocks that changed since that backup.
+2. System prepares VM for backup for running VM, or the VM disks for
+backup for a stopped VM.
+3. Backup application starts image transfer for download
+4. Backup application obtains changed blocks list from ovirt-imageio
+5. Backup application downloads changed blocks from ovirt-imageio
+6. Backup application finalize image transfer
+7 Backup application finish backup using oVirt API
+
+
+## Create VM with incremental backup support
+To allow incremental backup the user must enable the feature per VM.
+
+### Open issues
+Do we need to support per disk incremental backup?
+
+
+## Restoring from incremental backup
+Basically the backup vendor will build a snapshot image from the
+incremental backups, and upload the snapshot using imageio. No change
+in libvirt or qemu is needed.
+
+Flow:
+1. User select restore point based on available backups using the
+backup application (not part of ovirt).
+2. Backup application builds the complete snapshot from the available
+backups.
+3. Backup application upload the complete snapshot to storage using
+imageio.
+
+### Open issues
+Do we have a need for incremental restore, modifying specific blocks
+in the a snapshot from incremental backups?
+
+
+## Preparing running vm for incremental backup
+Qemu supports now only the push mode - copying changed blocks to
+external disk provided by the user, using qcow2 format. Theoretically
+any format can be used but practically only qcow2 allows getting
+changed blocks on any storage type.
+
+A more efficient pull mode is being developed, when the user copy
+changed blocks from the active layer. In this mode qemu may need to
+copy only blocks that the running VM want to change after the
+backup started.
+
+Since the pull mode is unlikely to be available in next RHEL
+version (7.6) we plan to use the push mode now, and use the pull
+mode when it is available.
```

Comment:
We were informed that both push and pull models will either be in or not in RHEL 7.6, since they both use the same libvirt API and both are done in QEMU.